services/pull/pull.go

```diff
@@ -48,6 +48,15 @@ func NewPullRequest(ctx context.Context, repo *repo_model.Repository, issue *iss
 		return user_model.ErrBlockedUser
 	}
 
+	// user should be a collaborator or a member of the organization
+	if !issue.Poster.IsAdmin {
```

Comment:
I don't think the logic is right. https://github.com/go-gitea/gitea/pull/31033#issuecomment-2124966661